ocaml/xapi/xapi_pool.ml

```diff
@@ -626,6 +626,138 @@ let create_or_get_pif_on_master __context rpc session_id (pif_ref, pif) : API.re
 
   new_pif_ref
 
+let copy_management_vlan_from_master __context rpc session_id mgmt_interface : API.ref_VLAN =
+
+  let vlan_pif_device = ref "" in
+  let remote_tagged_pif_network = ref Ref.null in
+
+  (* Get the new Host ref for the joining Host *)
+  let pif = Db.PIF.get_record ~__context ~self:mgmt_interface in
+  let my_host_ref = pif.API.pIF_host in
+  let my_host = Db.Host.get_record ~__context ~self:my_host_ref in
+  let new_host_ref = create_or_get_host_on_master __context rpc session_id (my_host_ref, my_host) in
+
+  (* Get the vlan record *)
+  let vlan_ref, vlan_record = List.hd (Db.VLAN.get_records_where ~__context ~expr:
+      (Eq (Field "untagged_PIF", Literal (Ref.string_of mgmt_interface))))
+  in
+
+  (* Returns the new PIF ref on same network as remote PIF for the joining Host *)
+  let get_my_new_physical_pif remote_pif =
+    let network = Client.PIF.get_network rpc session_id remote_pif in
+    let expr =
+      Printf.sprintf "field \"network\"=\"%s\" and field \"host\"=\"%s\""
+      (Ref.string_of network)
+      (Ref.string_of new_host_ref)
+    in
+    begin match Client.PIF.get_all_records_where ~rpc ~session_id ~expr with
+    | [] -> Ref.null
+    | (pif, _) :: _ -> pif
+    end
+  in
+
+  (* Get the remote management PIF config, VLAN config and Bond config *)
+  let remote_mgmt_pif = Client.Host.get_management_interface rpc session_id (get_master ~rpc ~session_id) in
+  let remote_mgmt_network = Client.PIF.get_network rpc session_id remote_mgmt_pif in
+  let remote_mgmt_vlan = Client.PIF.get_VLAN_master_of rpc session_id remote_mgmt_pif in
+  let remote_tagged_pif = Client.VLAN.get_tagged_PIF rpc session_id remote_mgmt_vlan in
+  let remote_mgmt_bond = Client.PIF.get_bond_master_of rpc session_id remote_tagged_pif in
+
+  (* If management VLAN exists on a Bond on Pool then copy the bond from master for joining Host *)
```

Comment:
I know I suggested this earlier, but I now think that it is better to keep this simple, and let the "synchronise bonds" startup item deal with this when xapi restart as a slave. What we should do here, is recreate the VLAN on the PIF it is currently on, and put it on the network that the pool master's management PIF is on. On restart, the bond will be created and the VLAN and management settings will be automatically moved to it.